Client/Frontend/Widgets/CurveView.swift

```diff
@@ -0,0 +1,63 @@
+/* Code for drawing the urlbar curve */
```

Comment:
I'm not really worried about us breaking encapsulation in this case; I'd mostly just prefer it stay in `URLBarView` for organization reasons. It seems nicer to have one fewer class in the global namespace, and one fewer file in Widgets/. It just adds clutter if we put things in Widgets/ that, well...aren't widgets :)

To flip your "really gains us anything" statement around, the curve is already committed in the repo as a private class in `URLBarView`. Do we gain anything by introducing a change to pull it out?